File: run_all.R
# Run all R analyses.

args <- commandArgs(trailingOnly = TRUE)
script_arg <- commandArgs(trailingOnly = FALSE)
file_arg <- sub("^--file=", "", script_arg[grepl("^--file=", script_arg)])
here <- if (length(file_arg)) dirname(normalizePath(file_arg[1])) else getwd()
data_arg <- if (length(args)) args[1] else ""

scripts <- c(
  "Table_1.R",
  "Table_2.R",
  "Table_3.R",
  "Table_4.R",
  "Figure_1.R",
  "Figure_2.R",
  "Supplementary_Table_S1.R",
  "Supplementary_Table_S2.R",
  "Supplementary_Table_S3.R",
  "Supplementary_Table_S4.R",
  "Supplementary_Table_S5.R",
  "Supplementary_Table_S6.R",
  "Supplementary_Figure_S1.R",
  "Supplementary_Figure_S2.R",
  "Supplementary_Figure_S3.R"
)

for (script in scripts) {
  cat("\n===", script, "===\n")
  cmd <- c(file.path(here, script), if (nzchar(data_arg)) data_arg)
  status <- system2("Rscript", cmd)
  if (status != 0) stop(paste("Failed:", script))
}

File: run_all.py
# Run all Python analyses.

from pathlib import Path
import subprocess
import sys

here = Path(__file__).resolve().parent
scripts = [
    "Table_1.py",
    "Table_2.py",
    "Table_3.py",
    "Table_4.py",
    "Figure_1.py",
    "Figure_2.py",
    "Supplementary_Table_S1.py",
    "Supplementary_Table_S2.py",
    "Supplementary_Table_S3.py",
    "Supplementary_Table_S4.py",
    "Supplementary_Table_S5.py",
    "Supplementary_Table_S6.py",
    "Supplementary_Figure_S1.py",
    "Supplementary_Figure_S2.py",
    "Supplementary_Figure_S3.py",
]

data_arg = sys.argv[1] if len(sys.argv) > 1 else None

for script in scripts:
    cmd = [sys.executable, str(here / script)]
    if data_arg:
        cmd.append(data_arg)
    print(f"\n=== {script} ===")
    subprocess.run(cmd, check=True)
